### PR TITLE
Archive artifacts to build.openmodelica.org

### PR DESCRIPTION
**Pull request by AnHeuermann**
Upload artifacts to build.openmodelica.org for Pythons pip installation of OMSimulator python module.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -465,17 +472,89 @@ EXIT /b 1
...
+            unstash name: 'mingw64-zip'       // includes: "OMSimulator-mingw64*.zip"
+            unstash name: 'win64-zip'         // includes: "OMSimulator-win64*.zip"
+            unstash name: 'osx-zip'           // includes: "OMSimulator-osx*.zip"
+
+            sh "ls *.zip *.tar.gz"
+
+            sshPublisher (
+              publishers: [
+                sshPublisherDesc(
+                  configName: 'OMSimulator-doc',
```
Comment (adrpo) [inline Jenkinsfile:531]: This needs to be "OMSimulator" instead "OMSimulator-doc".

Comment (AnHeuermann) [reply Jenkinsfile]: Changed it.

**Comment by AnHeuermann** [comment]
Now the artifacts will only be uploaded if the Jenkins build get's started with `UPLOAD_BUILD_OPENMODELICA`. We should only upload "good" builds, e.g. tagged versions or if a new feature is available that the Python version needs.

**Comment by sjoelund** [comment]
You can check in Jenkins if you are building a pull request, a branch, or a tag and only upload automatically based on those.

**Comment by AnHeuermann** [comment]
> You can check in Jenkins if you are building a pull request, a branch, or a tag and only upload automatically based on those.

Yeah. But I don't want to upload artifacts for all successful builds on master. To many files with only a few changes. The lastest installs will always be available as Jenkins artifacts. But we probably want to have more then only the tagged commits.
At least this is my opinion.

**Comment by sjoelund** [comment]
> Yeah. But I don't want to upload artifacts for all successful builds on master. To many files with only a few changes. The lastest installs will always be available as Jenkins artifacts. But we probably want to have more then only the tagged commits.

You can filter out master :)

**Review by adrpo — APPROVED** [review]
OK by me!

**Review by lochel — CHANGES_REQUESTED** [review]
If I get it right, the final download link would be something like this:

    https://build.openmodelica.org/omsimulator/linux/i386/OMSimulator-linux-amd64-v2.0.0-242-ga90fcb4.tar.gz

I would like to change the naming and directory structure to something like this:

    https://build.openmodelica.org/omsimulator/linux-i386/OMSimulator-2.0.0.post245.tar.gz

**Comment by sjoelund** [comment]
> I would like to change the naming and directory structure to something like

Please follow semver.org for names so they can be sorted properly. Dots after 2.0.0 are a no-no. Never put a name like `post245`, but rather use `post.245`.

**Comment by lochel** [comment]
> Please follow semver.org for names so they can be sorted properly. Dots after 2.0.0 are a no-no. Never put a name like `post245`, but rather use `post.245`.

No, we don't use that version schema. You can open an issue to discuss changing the version schema if you want. But please don't clutter this PR.

**Comment by AnHeuermann** [comment]
Changing the directories to `omsimulator/linux-i386/` (and analogous for the others) is no problem. I'll add a commit for that.

Changing `OMSimulator-linux-amd64-v2.0.0-242-ga90fcb4.tar.gz` has all sort of implications. We should have only one format for OMSimulator version (something like `OMSimulator v2.0.0.post236-ga34ab0e-linux`), the name of the archive we save on Jenkins for the latest install and the installs we want to save on build.openmodelica.org.

So what is the format we wan't to have for this? Whatever it is we need to change it at some additional places (e.g. FMI Cross Check on Jenkins).

But I would say let's only make the first change in this PR and discuss how the name for version and artifacts should look like an change that there.

File: Jenkinsfile [hunk 2]
```diff
@@ -465,17 +473,90 @@ EXIT /b 1
...
+                      sourceFiles: 'OMSimulator-mingw32*.zip'),
+                    sshTransfer(
+                      remoteDirectory: "win-mingw64/",
+                      sourceFiles: 'OMSimulator-mingw64*.zip'),
+                    sshTransfer(
+                      remoteDirectory: "osx/",
+                      sourceFiles: 'OMSimulator-osx*.zip'),
+                    sshTransfer(
+                      remoteDirectory: "win-msvc64/",
+                      sourceFiles: 'OMSimulator-win64*.zip')
```
Comment (lochel) [inline Jenkinsfile:555]: Do we need to create the directories on build.openmodelica.org manually?

Comment (sjoelund) [reply Jenkinsfile:555]: Not if you create the directory locally and transfer the `sourceFiles: 'linux-i386/OMSimulator-linux-i386-*.tar.gz'` I think. Also note that it's not `i368`

Comment (AnHeuermann) [reply Jenkinsfile]: The ssh transfer creates all directories. At least it did when I uploaded to the wrong location.

**Review by lochel — APPROVED** [review]

**Pull request merged**